cmd/aries-agentd/startcmd/start_test.go

```diff
@@ -0,0 +1,304 @@
+	}
+
+	requiredFlagAnnotation, keyExists := flagAnnotations["cobra_annotation_bash_completion_one_required_flag"]
+	if len(requiredFlagAnnotation) != 1 {
+		return false
+	}
+	if keyExists {
+		if requiredFlagAnnotation[0] != "true" {
+			return false
+		}
+	} else {
+		return false
+	}
+
+	return true
+}
+
+func TestStartAriesD(t *testing.T) {
+	// TODO - remove this path manipulation after implementing #175 and #148
+	path, cleanup := generateTempDir(t)
+	defer cleanup()
+
+	go startAgent(testHostURL, testInboundHostURL, path)
+
+	validateRequests(t)
+}
+
+// TODO: this method should be refactored (e.g., too many lines).
+//nolint:funlen
+func validateRequests(t *testing.T) {
+	newreq := func(method, url string, body io.Reader, contentType string) *http.Request {
+		r, err := http.NewRequest(method, url, body)
+		if contentType != "" {
+			r.Header.Add("Content-Type", contentType)
+		}
+		if err != nil {
+			t.Fatal(err)
+		}
+		return r
+	}
+
+	tests := []struct {
+		name               string
+		r                  *http.Request
+		expectedStatus     int
+		expectResponseData bool
+	}{
+		// controller API test
+		{
+			name:               "1: testing get",
+			r:                  newreq("GET", fmt.Sprintf("http://%s/connections/create-invitation", testHostURL), nil, ""),
+			expectedStatus:     http.StatusOK,
+			expectResponseData: true,
+		},
+
+		// DIDComm inbound API test
+		{
+			name: "200: testing didcomm inbound",
+			r: newreq("POST",
+				fmt.Sprintf("http://%s", testInboundHostURL),
+				strings.NewReader(`
+							{
+								"@id": "5678876542345",
+								"@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/didexchange/1.0/invitation"
+							}`),
+				"application/didcomm-envelope-enc"),
+			expectedStatus:     http.StatusInternalServerError,
+			expectResponseData: false,
+		},
+	}
+
+	// give some time for server to start
+	// TODO instead of sleep, listen for port
+	time.Sleep(100 * time.Millisecond)
+
+	for _, tt := range tests {
+
+		resp, err := http.DefaultClient.Do(tt.r)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		defer func() {
+			e := resp.Body.Close()
+			if e != nil {
+				panic(err)
+			}
+		}()
+
+		require.Equal(t, tt.expectedStatus, resp.StatusCode)
+		if tt.expectResponseData {
+			require.NotEmpty(t, resp.Body)
+			response, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				t.Fatal(err)
+			}
+			require.NotEmpty(t, response)
+			require.True(t, isJSON(response))
+		}
+	}
+
+}
+
+// isJSON checks if response is json
+func isJSON(res []byte) bool {
+	var js map[string]interface{}
+	return json.Unmarshal(res, &js) == nil
+
+}
+
+func TestStartCmdWithoutHostArg(t *testing.T) {
+	done := make(chan bool)
+
+	go func() {
+		startCmd := Cmd()
+		args := []string{"--" + AgentInboundHostFlagName, testInboundHostURL, "--" + AgentDBPathFlagName, ""}
+		startCmd.SetArgs(args)
+		startCmd.Execute()
+		done <- true
+	}()
+
+	select {
+	case res := <-done:
+		require.True(t, res)
+	case <-time.After(5 * time.Second):
```

Comment:
Talked to Bob and he suggested that I can mock the server instead of letting it actually launch here.

Reply:
This suggestion came up because I was asking how I can test a successful server start without causing it to block. Running the server in another goroutine causes issues too since the goroutine lives on for other tests and causes them to fail.